components/GeneralComponents/Buttons/index.js

```diff
@@ -0,0 +1,109 @@
+import styled from "styled-components";
+import Image from "next/image";
+import TrashIcon from "../../../public/trash-can.svg";
+import ArchiveIcon from "../../../public/box-open.svg";
+import EditIcon from "../../../public/pen-to-square.svg";
+import BackIcon from "../../../public/back-icon.svg";
+import AddIcon from "../../../public/add-item.svg";
+import SubmitIcon from "../../../public/paper-plane.svg";
+import { useState } from "react";
+import Spinner from "../CircleAnimation";
+import {
+  SuccessCreateMessage,
+  SuccessUpdateMessage,
+  DeleteMessage,
+  ArchiveMessage,
+} from "../Toasts";
+import "react-toastify/dist/ReactToastify.css";
+
+export function StyledButtonComponent({
+  children,
+  type,
+  callback,
+  item,
+  disabled,
+}) {
+  function getIcon(type) {
+    if (type === "edit") {
+      return EditIcon;
+    } else if (type === "back") {
+      return BackIcon;
+    } else if (type === "delete") {
+      return TrashIcon;
+    } else if (type === "archive") {
+      return ArchiveIcon;
+    } else if (type === "add") {
+      return AddIcon;
+    } else if (type === "update") {
+      return SubmitIcon;
```

Comment:
You can use an object as a "map" for such things, e.g.:

```js
const buttonIcons{
   edit:  EditIcon,
   back: BackIcon,
   delete: TrashIcon,
   // ...
}
```
you can then access the icon with bracket notation:
```js
   <Image src={buttonIcons[type]} />
```